base/client.jl

```diff
@@ -169,24 +169,50 @@ function _parse_input_line_core(s::String, filename::String)
     return ex
 end
 
-function parse_input_line(s::String; filename::String="none", depwarn=true)
+"""
+    parse_input_line(s::AbstractString; filename::AbstractString = "none", depwarn::Bool = true)
+
+Parse the contents of `s` as Julia code.
+If `filename` is provided, it will be used as the filename
+in all `LineNumberNode`s.
+If `depwarn` is `false`, deprecation warnings will be suppressed.
+
+```julia
+julia> expr = parse_input_line("c = sqrt(a^2 + b^2)")
+```
+
+```julia
+julia> filename = "hello.jl"
+
+julia> expr = parse_input_line(read(filename, String); filename = filename)
+```
+"""
+function parse_input_line(s::String; filename::AbstractString = "none", depwarn::Bool = true)
     # For now, assume all parser warnings are depwarns
     ex = if depwarn
-        _parse_input_line_core(s, filename)
+        _parse_input_line_core(s, String(filename))
```

Comment:
I'd be inclined to leave this conversion to `_parse_input_line_core` which is the location where the concrete type is required.

Reply:
In the new `Meta.parsefile`, I kept this conversion in. The reason is that if we don't convert `filename` to a `String`, then if e.g. you pass a `SubString{String}` as the `filename`, when you try to call `parse_input_line` you get `ERROR: TypeError: in keyword argument s, expected String, got SubString{String}`.